tests/integration/helpers.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import io
+import os
+import pathlib
+import uuid
+import zipfile
+
+import yaml
+
+
+def inject_venv(charm: pathlib.Path | str, src: pathlib.Path | str):
+    """Inject a Python library into the charm venv directory inside a charm file."""
+    zip_file = zipfile.ZipFile(charm, "a")
```

Comment:
I'm wondering if this can benefit from `with zipfile.Zipfile` context :)

Reply:
Yes, indeed that's safer, updated, thanks!